Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/GazePointerStateMachineTests.cs

```diff
@@ -34,16 +34,122 @@ public void TestHandAndSpeechBehaviour()
             gsm.OnSpeechKeywordRecognized(data);
             Assert.IsFalse(gsm.IsGazePointerActive, "After select is called but hands are up, gaze pointer should not show up");
 
-            gsm.UpdateState(0, 0);
+            gsm.UpdateState(0 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, false);
             gsm.OnSpeechKeywordRecognized(data);
-            gsm.UpdateState(0, 0);
+            gsm.UpdateState(0 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, false);
             Assert.IsTrue(gsm.IsGazePointerActive, "When no hands present and select called, gaze pointer should show up");
 
             // Say select while gaze pointer is active, then raise hand. Gaze pointer should go away
             gsm.OnSpeechKeywordRecognized(data);
-            gsm.UpdateState(1, 0);
-            gsm.UpdateState(1, 0);
+            gsm.UpdateState(1 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, false);
+            gsm.UpdateState(1 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, false);
             Assert.IsFalse(gsm.IsGazePointerActive, "After select called with hands present, then hand up, gaze pointer should go away");
         }
+
+        [Test]
+        public void TestEyeGazeHandAndSpeechBehaviour()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene(true);
+
+            // Initial state: gaze pointer active
+            var gsm = new GazePointerVisibilityStateMachine();
+            Assert.IsTrue(gsm.IsGazePointerActive, "Gaze pointer should be visible on start");
+
+            // With the hand raised, eye gaze pointer should still exist because only far interaction causes the 
+            // eye gaze pointer to go away.
+            gsm.UpdateState(1 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, true);
+            Assert.IsTrue(gsm.IsGazePointerActive, "With near interaction, gaze pointer should continue to exist");
+
+            // With far interaction active, eye gaze pointer should be hidden.
+            gsm.UpdateState(0 /*numNearPointersActive*/, 1 /*numFarPointersActive*/, true);
+            Assert.IsFalse(gsm.IsGazePointerActive, "With far interaction, gaze pointer should go away");
+
+            // Reset the state and validate that it goes back to being visible.
+            gsm.UpdateState(0 /*numNearPointersActive*/, 0 /*numFarPointersActive*/, true);
+            Assert.IsTrue(gsm.IsGazePointerActive, "Gaze pointer should be visible when no near or far pointers");
+
+            // Saying "select" should have no impact on the state of eye gaze-based interactions.
+            SpeechEventData data = new SpeechEventData(EventSystem.current);
+            data.Initialize(new BaseGenericInputSource("test input source", new IMixedRealityPointer[0], InputSourceType.Voice),
```

Comment:
Maybe pull into utility? We've pasted this a few times now.

Reply:
Put into a static private, utility is overkill until we need to use it in multiple places.